app/controllers/chat_channels_controller.rb

```diff
@@ -240,6 +241,25 @@ def retry_archive
     render json: success_json
   end
 
+  def toggle_open_status
```

Comment:
Something to consider here is whether we should just let the client tell us what state it wants the channel to be in. Toggling assumes the client's impression of the state and the server's state is the same which might not always be the case. I've been bitten by this in the past with regards to the toggling of status for topics.

Reply:
Yeah thanks for the context, I will change to dedicated endpoints.

Reply:
Instead of `toggle_state`, how about `change_state`? We can keep it as a single endpoint in that case.

Reply:
Yeah makes sense, good idea :+1: